types.go

```diff
@@ -972,7 +994,7 @@ func (f *Float64) MarshalJSON() ([]byte, error) {
 		return []byte("\"nan\""), nil
 	default:
 	}
-	return json.Marshal(float64(*f))
+	return json.Marshal(fmt.Sprintf("%f", float64(*f)))
```

Comment:
One big point to propose my another `Float` and the problem(assertion in different types) could be solved easier than I expected, and I believe that this change would have no problem in backward compatiblity

As-was
```
Diff:
                                --- Expected
                                +++ Actual
                                @@ -243,3 +243,3 @@
                                   (string) (len=8) "is_proxy": (float64) 0,
                                -  (string) (len=16) "last_vote_weight": (string) (len=19) "697460718699.945435",
                                +  (string) (len=16) "last_vote_weight": (float64) 6.974607186999454e+11,
                                   (string) (len=5) "owner": (string) (len=12) "teamgreymass",
                                @@ -247,3 +247,3 @@
                                   },
                                -  (string) (len=19) "proxied_vote_weight": (string) (len=8) "0.000000",
                                +  (string) (len=19) "proxied_vote_weight": (float64) 0,
```

Reply:
Seems like even worse breaking change IMO (although I like the general ideal), because now people marshalling will "silently" receive a `string` type instead of a `number` type in JSON.

Meaning you upgrade `eos-go` and marshalling type is different meaning consuming code needs to be updated (or you need hardest work to come back to it.

I have an in-between solution although it's imperfect. We could have a global variable that controls this (string vs number marshalling) and by default would stay with the old behavior but in the tests we would override it to print to string.

So, we have 3 possibilities here:
- Status quo (tests uses fixed value that are always precise)
- Override variable for testing purposes
- Breaking change `number` -> `string` but highly documented

Reply:
I added a global value  the overriding plan here's my push
https://github.com/eoscanada/eos-go/pull/191/commits/ac0a8d441e670b415cb56963113f4eb1cd91a5ac